tests/contrib/celery/test_utils.py

```diff
@@ -162,9 +163,81 @@ def test_task_id_from_protocol_v1():
     assert task_id == "dffcaec1-dd92-4a1a-b3ab-d6512f4beeb7"
 
 
-def test_task_id_from_protocol_v2():
+def test_task_id_from_protocol_v1_with_headers():
+    # ensures a `task_id` is properly returned when Protocol v1 is used with headers.
+    # `context` is an example of an emitted Signal with Protocol v1
+    # this tests ensures that the headers have other information
+    context = {
+        "body": {
+            "expires": None,
+            "utc": True,
+            "args": ["user"],
+            "chord": None,
+            "callbacks": None,
+            "errbacks": None,
+            "taskset": None,
+            "id": "dffcaec1-dd92-4a1a-b3ab-d6512f4beeb7",
```

Comment:
nit (not required): make the task ID a variable

Reply:
done in the newest commit!